app/assets/javascripts/global_activities/index.js

```diff
@@ -0,0 +1,56 @@
+/* global animateSpinner */
+
+'use strict';
+
+(function() {
+  function initExpandCollapseAllButtons() {
+    $('#global-activities-colapse-all').on('click', function() {
+      $('.activities-group').collapse('hide');
+    });
+    $('#global-activities-expand-all').on('click', function() {
+      $('.activities-group').collapse('show');
+    });
+  }
+
+  function initExpandCollapseButton() {
+    $('.activities-group').on('hide.bs.collapse', function() {
+      $('#' + $(this)
+        .attr('id') + '-button').find('.fas').removeClass('fa-caret-down');
+      $('#' + $(this)
+        .attr('id') + '-button').find('.fas').addClass('fa-caret-right');
+    });
+    $('.activities-group').on('show.bs.collapse', function() {
+      $('#' + $(this)
+        .attr('id') + '-button').find('.fas').removeClass('fa-caret-right');
+      $('#' + $(this)
+        .attr('id') + '-button').find('.fas').addClass('fa-caret-down');
```

Comment:
Maybe simplify it by adding data attribute to element and use add/remove class action in one line:

$(this.dataset.buttonLink).find('.fas').removeClass('fa-caret-down').addClass('fa-caret-right')

Reply:
Changed